### ISSUE TITLE
Lab 05 Testcase 9_03

### ISSUE THREAD

**Issue by aarontxz** [nus-cs2030/2223-s2#388]
# Summary

Hello, any idea why 19 can wait at 2 but then when 20 arrives it cannot wait at 2, I am pretty sure I checked the queue length properly if not I won't have been able to pass lab3, what other reason can cause 19 to be able to wait at 2 but 20 to leave.

when a server is resting does the queue move forward to free up a space for 20 to wait?

<img width="388" alt="Screenshot 2023-03-13 at 8 01 20 PM" src="https://user-images.githubusercontent.com/115057223/224699507-a9eda83e-38f0-48ae-b1cc-cb74cc48b70c.png">

**Comment by hhuilingg** [comment]
yes, the customer queue will be shift forward once the customer is serve. for example: when customer A is in first in queue and customer B is second in queue. when the serve is not busy, customer A will be in serve. at the this the customer queue will also be updated such that customer B is the first in queue and there will be spaces in queue for customer C to join.

(note that even customer A has not done yet, the queue is already updated)

**Comment by e0968868** [comment]
when a server is resting, the queue doesnt move forward to free up a space 
scenario: qmax is 2, customers 2 and 3 are in the queue, server is serving customer 1. when server finishes serving customer 1, customer 2 can only move forward and get served when the server has finished resting. i.e. while the server is resting, customer 4 can't join the queue as it's filled by customers 2 and 3, so customer 4 has to leave. to make this happen, i only removed the customer at my rest event, but i guess it depends on your own implementation

**Comment by hhuilingg** [comment]
yeap seems like it depends on your own implementations as my way works too :)

**Comment by nigel-tannn** [comment]
Would recommend having a separate event where the server is done resting and is ready to serve again, really helped me with this part!

**Comment by edwinhrj** [comment]
the queue doesn't move forward when the server is resting, and only moves forward by 1 customer when the server has just finished resting and is able to serve the first customer in its queue - thereby allowing the server to take in another customer in its queue.
